src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -522,18 +533,30 @@ public static JsonTypeInfo<T> CreateJsonTypeInfo<T>(JsonSerializerOptions option
                 ThrowHelper.ThrowArgumentNullException(nameof(options));
             }
 
-            DefaultJsonTypeInfoResolver.RootDefaultInstance();
-            return new CustomJsonTypeInfo<T>(options);
+            JsonConverter converter = DefaultJsonTypeInfoResolver.GetConverterForType(typeof(T), options, resolveJsonConverterAttribute: false);
+            return new CustomJsonTypeInfo<T>(converter, options);
         }
 
         private static MethodInfo? s_createJsonTypeInfo;
 
         /// <summary>
-        /// Creates JsonTypeInfo
+        /// Creates a blank <see cref="JsonTypeInfo"/> instance.
         /// </summary>
-        /// <param name="type">Type for which JsonTypeInfo stores metadata for</param>
-        /// <param name="options">Options associated with JsonTypeInfo</param>
-        /// <returns>JsonTypeInfo instance</returns>
+        /// <param name="type">The type for which contract metadata is specified.</param>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> instance the metadata is associated with.</param>
+        /// <returns>A blank <see cref="JsonTypeInfo{T}"/> instance.</returns>
```

Comment:
might be worth mentioning that `T` is matching passed in `type`

Reply:
Actually that seems wrong since this is not a generic method. I'll update it.

Reply:
technically runtime type is still generic, either way needs some clarifyin